app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,26 @@
+class FriendshipsController < ApplicationController
+  before_action :require_login
+  
+  def index 
+    @friendships = current_user.following_friendships 
+  end
+  
+  def create
+    leader = User.find(params[:leader_id])
+    
+    if current_user.follows?(leader)
+      flash[:danger] = "You already follow #{leader.full_name}!" 
+      redirect_to people_path
+    else 
+      Friendship.create(leader: leader, follower: current_user) unless leader == current_user
+      redirect_to people_path
```

Comment:
I would move this redirect down out of the `if/else`  since it runs either way. I think you can leave how you did the `unless leader...` different than `current_user.follows?...` because the first breaks out to do something, but the second simply fails to create--I think that flow works well.